src/UI/Components/Diagram/Canvas.tsx

```diff
@@ -54,6 +55,21 @@ const Canvas = ({
     service: mainServiceName,
   });
 
+  const handleLooseEmbeddedEvent = (event) => {
+    const customEvent = event as CustomEvent;
+    const eventData: { kind: "remove" | "add"; id: string } = JSON.parse(
+      customEvent.detail,
+    );
+    if (eventData.kind === "remove") {
+      setLooseEmbedded((prevSet) => {
```

Comment:
you are returning a new Set to avoid mutating the previous state, but by doing the delete action, you are mutating. 
```
setLooseEmbedded((prevSet) => {
  const newSet = new Set(prevSet);
  newSet.delete(eventData.id);
  return newSet;
});
```
You can do it this way to avoid mutating the previous state.